add-on/src/lib/ipfs-companion.js

```diff
@@ -274,19 +276,25 @@ module.exports = async function init () {
     }
   }
 
-  function uploadResultHandler (result) {
-    result.forEach(function (file) {
+  async function uploadResultHandler (result) {
+    for (let file of result) {
       if (file && file.hash) {
         const {path, url} = getIpfsPathAndNativeAddress(file.hash)
-        browser.tabs.create({
-          'url': url
-        })
-        console.info('[ipfs-companion] successfully stored', path)
+        console.log('uploadResultHandler.file-' + result.indexOf(file), file)
+        // open the wrapping directory (or the CID if wrapping was disabled)
+        if (result.length === 1 || file.path === '' || file.path === file.hash) {
+          await browser.tabs.create({
```

Comment:
What's the expected behaviour here? If i add a single file and let it be wrapped, I'm seeing it open the directory not the file. If i add multiple, it shows the directory.

Reply:
Yes, that is correct.

If wrapping is enabled or you are uploading more than one file, it will **always open wrapping directory** (the default).
If wrapping is disabled and a single file is uploaded, it will be opened directly. (requires explicit opt-in from user via upload options).
 
I feel it is a good default for now:
- Most of users want to keep file names and as a side effect will get a branded directory listing provided by gateway
  (Until we integrate ideas from https://github.com/tableflip/share-via-ipfs via separate PRs)
- Uploaded file can be a `.zip` file and browser can't show that. 
  (Opening the file itself will just close the tab and start the download, which makes it difficult for user to copy the URL)
- Mobile users are not forced to download the data, they just get directory listing and decide what do do with it (open the specific file, copy link to directory, copy direct link to file etc)